docs/github-provider.md

```diff
@@ -0,0 +1,109 @@
+# How to use the *GitHub Provider*
+
+## Setup the provider
+
+To setup the provider you need to do 2 main things:
+- get an access token from your GitHub account
+- set the access token in your development environment
+
+ℹ️ You can also use the GitHub API without access token. But you will be limited in the number of request you can make.
+So you are free to skip this setup part untill you are limited by the api.
+
+### Access token
+
+- Go to https://github.com/settings/tokens/ and click on "Generate new token" button. (you can choose either a "Fine-grained token" or a "Classic token").
+- After generating the token, keep it preciously.
+
+### Set the token
+
+- Create a new file named ```.env``` at the root of the project and write this line: ```export GITHUB_TOKEN="<token_generated>"```. 
+  Replace ```<token_generated>``` with the token you just copied from GitHub.
+- Run this command: ```source .env```.
+  This command allow you to export the ```GITHUB_TOKEN``` environment variable and use it in the Sismo Hub application.
+
+
+You can now use the GitHub Provider!
+
+
+---
+
+## Use the provider
+
+### Methods
+
+There are 2 main methods in the GitHub Provider:
+- ```getRepositoriesContributors```
+- ```getRepositoriesStargazers```
+
+<br>
+
+#### getRepositoriesContributors
+
+**This method allow you to fetch all the contributors of one or more GitHub repositories.**
+
+There are 3 arguments to give to this method:
+- ```repositories```
+- ```getOrganizationMembers```
+- ```defaultValue```
+
+#### getRepositoriesStargazers
+
+**This method allow you to fetch all the GitHub user who put a star on one or more GitHub repositories.**
+
+There are 2 arguments to give to this method:
+- ```repositories```
+- ```defaultValue```
+
+<br>
+
+#### repositories
+
+```repositories``` is the array that contains all the repository you want to fetch.
+For example, if you want to fetch all the contributors of the Solidity GitHub repository from Ethereum, you need to pass this argument: ```["ethereum/solidity"]```
+
+#### defaultValue
+
+```defaultValue``` define the value of all the items of the method returned object.
+The object returned by the methods is of this form:
+```
+{
+    "github:username:id":2,
+    "github:username:id":2
+}
+ ```
+So here, in this example, ```defaultValue``` is 2.
+By default ```defaultValue``` is 1.
+
+ℹ️ ```username``` and ```id``` corespond to the GitHub username and id of the user.
+
+#### getOrganizationMembers
+
+This concerns only the ```getRepositoriesContributors``` method.
+```getOrganizationMembers``` allow you to fetch the members of the organizations that own the repositories or not (in addition to the contributors). If ```{getOrganizationMembers: true}``` it will add the members, else if ```{getOrganizationMembers: false}``` it will not do it.
+
+<br>
+
+### Usage
+
+First you need to instantiate the GitHub Provider:
+```const githubProvider = new dataProviders.GithubProvider();```
```

Comment:
I personally prefer to put in a code block, it is cleaner for someone that only wants to see code blocks
```typescript
const githubProvider = new dataProviders.GithubProvider();
```